server/boot/user.js

```diff
@@ -266,6 +273,51 @@ module.exports = function(app) {
     return res.redirect('/' + username);
   }
 
+  function getUnlinkSocial(req, res, next) {
+    const { user } = req;
+    const { username } = user;
+    let social = req.params.social;
+    if (!social) {
+      req.flash('errors', {
+        msg: 'No social account found'
+      });
+      return res.redirect('/' + username);
+    }
+
+    social = social.toLowerCase();
+    const validSocialAccounts = ['twitter', 'linkedin'];
+    if (validSocialAccounts.indexOf(social) === -1) {
+      req.flash('errors', {
+        msg: 'Invalid social account'
+      });
+      return res.redirect('/' + username);
+    }
+
+    if (!user[social]) {
+      req.flash('errors', {
+        msg: `No ${social} account associated`
+      });
+      return res.redirect('/' + username);
+    }
+
+    const query = {};
+    query['userId'] = user.id;
+    query['provider'] = social;
+    return UserIdentity.destroyAll(query, function(err) {
+      if (err) { return next(err); }
+
+      const updateData = {};
+      updateData[social] = null;
+      user.update$(updateData);
```

Comment:
This is returns an observable. You must subscribe to it to start the update. Please review other places in the codebase where it is used.